### ISSUE TITLE
Constructor `readonly` properties not detected as a PHP 8.1 feature 

### ISSUE THREAD

**Issue by llaville** [llaville/php-compatinfo#374]
# Bug report category

- [x] Feature issue.

## Summary
Constructor readonly properties not detected as a PHP 8.1 feature.
Learn more with https://php.watch/versions/8.1/readonly#constructor-readonly

## Expected behaviour
Detect constructor `readonly` properties with `Bartlett\CompatInfo\Application\Sniffs\Classes\ReadonlyPropertySniff`

## Actual behaviour
Only class property in AST (with `Node\Stmt\Property`) is analysed. 
Should also analyse in AST (the param list of `Node\FunctionLike`)

**Comment by llaville** [comment]
Tests with https://github.com/llaville/php-compatinfo/releases/tag/7.1.3 on following source code.

```php
<?php
class Test {
    public function __construct(public readonly string $prop) {
    }
}
```

`analyser:run` command returns with console reporter

```
[OK] Requires PHP 8.0.0alpha1 (min)
```

and on details with rule reporter

```text
Stats
-----

 PHP 70 :   1 rule(s) found
 PHP 80 :   1 rule(s) found

Rules
-----

         code : CA7008
  description : Parameters Type Declaration 'string' is available since PHP 7.0.0
        sniff : ParamTypeDeclarationSniff
      methods : (1)
 Test\__construct

         code : CA8003
  description : Constructor property promotion is available since PHP 8.0.0
        sniff : PropertyPromotionSniff
      methods : (1)
 Test\__construct
```

While `CA8102` rule should also be detected

**Comment by llaville** [comment]
Verifications are now done. Closing this report

**Comment by llaville** [comment]
Fix included now in branch 7.2 for next feature release 7.2.0, PHP 8.4 compatible
